### PR TITLE
Con 4410 validate file format size when user chooses the file

### PR DESCRIPTION
**Pull request by kola-samuel**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Validators/Cohort/FileUploadStartViewModelValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            const string fileContents = "";
+            var textStream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(fileContents));
+            _file.Setup(m => m.OpenReadStream()).Returns(textStream);
+
+            var model = new FileUploadStartViewModel { Attachment = _file.Object };
+            AssertValidationResult(vm => vm.Attachment, model, false);
+        }
+
+        [Test]
+        public void ShouldReturnInvalidMessageWhenColumnCountIsNot13()
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Validators/Cohort/FileUploadStartViewModelValidatorTests.cs:61]: should we have separate unit tests for column count <13 and column count > 13?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/DependencyResolution/ConfigurationRegistry.cs [hunk 2]
```diff
@@ -20,6 +20,7 @@ public ConfigurationRegistry()
             AddConfiguration<ProviderFeaturesConfiguration>(ProviderCommitmentsConfigurationKeys.FeaturesConfiguration);
             AddConfiguration<ZenDeskConfiguration>(ProviderCommitmentsConfigurationKeys.ZenDeskConfiguration);
             AddConfiguration<DataProtectionConnectionStrings>(ProviderCommitmentsConfigurationKeys.DataProtectionConnectionStrings);
+            AddConfiguration<CsvConfiguration>(ProviderCommitmentsConfigurationKeys.CsvConfiguration);
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/DependencyResolution/ConfigurationRegistry.cs:23]: Would this be better name BulkUploadFileValidationConfiguration or something similarly specific?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        public FileUploadStartViewModelValidator(ILogger<FileUploadStartViewModelValidator> logger, CsvConfiguration csvConfiguration)
+        {
+            _logger = logger;
+            _csvConfiguration = csvConfiguration;
+
+            CascadeMode = CascadeMode.Stop;
+
+            RuleFor(x => x.Attachment)
+                .NotNull()
+                .Must(CheckFileSize).WithMessage("The selected file must be smaller than 50KB")
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs:29]: this message should use string interpolation to include the csvConfiguration.MaxFileSize

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                .Must(CheckFileType).WithMessage("The selected file must be a CSV")
+                .MustAsync(CheckEmptyFileContent).WithMessage("The selected file is empty")
+                .MustAsync(CheckFileColumnCount).WithMessage("The selected file could not be uploaded – use the template")
+                .MustAsync(CheckFileRowCount).WithMessage("The selected file must be less than 100 lines");
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs:33]: Same for this one

**Review by cofaulco — CHANGES_REQUESTED** [review]
a few comments and suggestions

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/DependencyResolution/ConfigurationRegistry.cs (hunk 2 above)
Comment (kola-samuel) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/DependencyResolution/ConfigurationRegistry.cs]: changed

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs (hunk 4 above)
Comment (kola-samuel) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs]: changed

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs (hunk 3 above)
Comment (kola-samuel) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Validators/FileUpload/FileUploadStartViewModelValidator.cs]: updated

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/FileUploadStart.cshtml [hunk 5]
```diff
@@ -8,7 +8,7 @@
 }
 
 <div class="govuk-grid-row">
-    <div class="govuk-grid-column-two-thirds">
+    <div class="govuk-grid-column-full">
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/FileUploadStart.cshtml:11]: can this line be remove? there is another grid-column-two-thirds within this div

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/FileUploadStart.cshtml [hunk 6]
```diff
@@ -32,3 +32,13 @@
 {
 <a id="back-link" class="govuk-back-link" href="@Url.Action("FileUploadInform", new { @Model.ProviderId })">Back</a>
 }
+
+
+<style type="text/css" >
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/FileUploadStart.cshtml:37]: don't think we should have this style section

Comment (kola-samuel) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Cohort/FileUploadStart.cshtml]: true

**Pull request merged**